GAWB-1693: bring autocomplete suggestions more in line with main text search

davidangb opened the pull request:
following Andrea's suggestion, I made the `autocomplete` analyzer look more like the `default` analyzer. This makes our suggestions more consistent with text search.

I also:
* changed autosuggest queries to be phrase search instead of regular search. This makes results for e.g. "brca o" more relevant; previously it would find anything with both "brca" and "o" anywhere in the text; now they need to be part of a phrase, next to each other.
* reduced our highlighter fragment size from 75 to 50. This sets the limit on the suggestion sizes. Entering "coho" will now suggest "This cohort is part of The Cancer Genome Atlas" instead of "This cohort is part of The Cancer Genome Atlas project (https" ... just a bit terser.


- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

davidangb committed:
bring autocomplete suggestions more in line with main text search

rushtong commented on src/test/scala/org/broadinstitute/dsde/firecloud/integrationtest/AutoSuggestSpec.scala:
```diff
@@ -0,0 +1,86 @@
+  }
+
+  final val HL_START = "<strong class='es-highlight'>"
+  final val HL_END = "</strong>"
```
Minor DRY fix - these two vals could be defined in `ElasticSearchDAOQuerySupport` and then re-used.

rushtong reviewed (approved):
Love the test. Also, good opportunity to read up on these ES features.

davidangb committed:
define highlighter tags centrally

davidangb committed:
reverting to allowing 1-char ngrams

davidangb committed:
autosuggest now uses a phrase query

davidangb commented:
@rushtong back to you for re-review. Since you reviewed last, I have addressed a use case of suggestions for "brca o".
* reverted the min ngram change; min ngram is back to 1. Otherwise, we wouldn't consider the "o" in "brca o"
* changed suggestion query to use phrase search. Previously, we found anything with both "brca" and "o" anywhere in the text; now they need to be next to each other.
* added a test for "brca o".
* removed the test for "tc", because I dropped the ngram size.
* addressed your feedback about the HL_START and HL_END variables - good catch!!!

rushtong reviewed (approved):
👍 Can't find anything to comment on ... looks great!
